packages/core/test/unit/html/linkProcessor.test.js

```diff
@@ -35,6 +35,36 @@ const mockConfig = {
   ],
 };
 
+test.each([
+  // Test converting .md to .html
+  ['.md', '', '', '', '<a href="/index.html">Test</a>'],
+  // Test converting .mbd to .html
+  ['.mbd', '', '', '', '<a href="/index.html">Test</a>'],
+  // Test conversion when both .md and .mbd are present, and .mbd is extension
+  ['.md', '.mbd', '', '', '<a href="/index.md.html">Test</a>'],
+  // Test conversion when both .md and .mbd are present, and .md is extension
+  ['.mbd', '.md', '', '', '<a href="/index.mbd.html">Test</a>'],
+  // Test converting .md to .html with URL fragment
+  ['.md', '', '#test-1', '', '<a href="/index.html#test-1">Test</a>'],
+  // Test converting .mbd to .html with URL fragment
+  ['.mbd', '', '#test-1', '', '<a href="/index.html#test-1">Test</a>'],
+  // Test conversion when both .md and .mbd are present, and .mbd is extension, with URL fragment
+  ['.md', '.mbd', '#test-1', '', '<a href="/index.md.html#test-1">Test</a>'],
+  // Test conversion when both .md and .mbd are present, and .md is extension, with URL fragment
+  ['.mbd', '.md', '#test-1', '', '<a href="/index.mbd.html#test-1">Test</a>'],
+  // Test no conversion when no-convert attribute is present
+  ['.md', '', '#test-1', 'no-convert', '<a href="/index.md#test-1" no-convert>Test</a>'],
+  // Test no conversion when neither .md nor .mbd extensions are present
+  ['', '', '', '', '<a href="/index">Test</a>'],
+])('Test link auto-conversion for (%s, %s, %s, %s)', (ext1, ext2, fragment, attr, expected) => {
```

Comment:
It's a little hard to come up with a proper test name that covers all the scenarios above. Do you think this is alright? Or should we aim to describe it more? @ang-zeyu

Reply:
this looks ok, especially since you've documented the test cases above.

Rather, it might be a little roundabout with the many parameters.

Let's compress the inputs into just one parameter (the input html) so the reader dosen't have to stitch them together, especially since all of them are rather short. Its also self-documenting (comments probably not needed in that case).

Reply:
Yep, that sounds a lot better! I initially separated them into different parameters as I thought it would make it more "modifiable" but in hindsight, it really is quite a roundabout (and quite unnecessary). 

But I think there's some value in keeping the comments, to make it more explicit and easier for the reader to quickly find out the semantics of the test case. What do you think? @ang-zeyu

Reply:
sure 👍